src/triangles.c

```diff
@@ -428,7 +428,6 @@ int igraph_i_trans4_al_simplify(igraph_adjlist_t *al,
 
 int igraph_transitivity_local_undirected4(const igraph_t *graph,
         igraph_vector_t *res,
-        const igraph_vs_t vids,
         igraph_transitivity_mode_t mode) {
```

Comment:
Does anyone fully understand the design of this function? Are we *sure* that there is no reason for, or benefit from having the four different versions of this function have the same signature?

This is one of those things I would not change without a deeper understanding. This change is literally aesthetic to silence a minor compiler warning that is not even on by default.

Reply:
This function is explicitly only called in the condition that all vertices are requested. To me that suggests that its intention is to never actually do something with this argument.

Instead, having the same arguments in all functions seems just esthetics to me. But if you find that they are somehow defined elsewhere to have the same call signature, I'd be happy to learn about it.

Reply:
I vaguely remember that Gábor has experimented with several different implementations of the transitivity function and that some of them later turned out to be efficient only if you need the transitivities of (almost) all vertices -- but then we did not bother removing the `vids` argument. There is no underlying deeper design decision here that would necessitate keeping the argument there, especially because these functions are meant to be internal anyway, so if later we somehow realize that they could still be made efficient for a subset of vertices only, we can still re-add the arguments without breaking the API (if we remove them from the public API now).

So, my vote is for removing the arguments and then renaming the functions to `igraph_i_...` and making them static.

Reply:
Yes, let us remove them from the headers in this PR instead of fussing with the develop branch. I checked and they are not used anywhere, so we will not need to create an internal header for them.

Reply:
If we remove this in this PR, then why not also remove the `*remove_duplicates` functions in this PR?

Reply:
We should rename the PR in that case to make this explicit (or at least add it in the PR description) -- but otherwise I'm fine with it.

Reply:
`igraph_transitivity_local_undirected4` was undocumented, and one might argue that it was fairly obvious that it was never meant for public use (i.e. it was exposed mistakenly). Therefore, removing it immediately makes sense.

Functions which were meant to be part of the public API are better deprecated for a version before removing them entirely.

Since all documented functions with `adjedgelist` in the name have been deprecated since 0.6, I support removing `igraph_adjedgelist_remove_duplicate` (even if it did not have an explicit deprecation notice).

----

As for `igraph_inclist_remove_duplicate`, I would not make changes to it in this PR, as there is no replacement functionality for it yet (i.e. add a flag controlling self-loop duplication to `inclist_init`). I think this one needs more discussion.

Reply:
Anyway, deciding whether we do the removal in this PR or later should definitely not block merging this PR! If there's a doubt, let's leave it for later and go ahead with this PR.

Reply:
OK, let's otherwise clean up those things in separate PRs, that might make it more targeted. I will undo the removal of the argument for now then, and in a next PR I will remove the deprecated `adjedgelist` functions, and in yet another we do something with `inclist_remove_duplicates`.